plugins/inputs/bigbluebutton/bigbluebutton.go

```diff
@@ -0,0 +1,167 @@
+package bigbluebutton
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type BigBlueButton struct {
+	URL         string `toml:"url"`
+	APIEndpoint string `toml:"api_endpoint"`
+	SecretKey   string `toml:"secret_key"`
+	ServerName  string `toml:"server_name"`
+}
+
+var bbbConfig = `
+	## Required BigBlueButton server url
+	url = "http://localhost:8090"
+
+	## Required BigBlueButton api endpoint
+	api_endpoint = "/bigbluebutton/api/"
+
+	## Required BigBlueButton secret key
+	# secret_key =
+
+	## Server name. Used as "server" tag if not empty. Otherwise use url
+	# server_name = 
+`
+
+func (b *BigBlueButton) SampleConfig() string {
+	return bbbConfig
+}
+
+func (b *BigBlueButton) Description() string {
+	return "Gather BigBlueButton web conferencing server metrics"
+}
+
+func (b *BigBlueButton) Gather(acc telegraf.Accumulator) error {
+	if b.SecretKey == "" {
+		return fmt.Errorf("BigBlueButton secret key is required")
+	}
+
+	if b.ServerName == "" {
+		b.ServerName = b.URL
+	}
```

Comment:
Please do this in an `Init()` function. This `Init()` function is called automatically by telegraf _once_ the config options are parsed and set in your structure. Doing so, eases the understanding of the code (input validation), simplifies `Gather()` and avoids doing the same boring evaluations over and over in each gather cycle.